### PR TITLE
ARROW-9191: [Rust] Do not panic when milliseconds is less than zero as chrono can handle…

### PR DESCRIPTION
**Pull request by maxburke**
… these formats

### CODE REVIEW

File: rust/parquet/src/record/api.rs [hunk 1]
```diff
@@ -893,16 +893,6 @@ mod tests {
...
-    #[test]
-    #[should_panic(expected = "Expected non-negative milliseconds when converting Int96")]
-    fn test_row_convert_int96_invalid() {
-        // INT96 value does not depend on logical type
-        let descr = make_column_descr![PhysicalType::INT96, LogicalType::NONE];
-
-        let value = Int96::from(vec![0, 0, 0]);
-        Field::convert_int96(&descr, value);
-    }
-
```
Comment (paddyhoran) [inline rust/parquet/src/record/api.rs:905]: Are there other tests we should be adding to test the effect of allowing negative milliseconds to flow through?  

I'm not that familiar with Parquet but I see things like [this](https://github.com/apache/arrow/blob/master/rust/parquet/src/record/api.rs#L584) which seem to indicate that we might be relying on milliseconds being non-negative.

Comment (maxburke) [reply rust/parquet/src/record/api.rs:905]: As a consumer of Parquet and Arrow I'm trying to not have my program panic when it encounters a date from 1968. 

Perhaps it's a mistake to have the timestamp types implemented with unsigned integers? Should these be signed?

Regardless, from an API design point of view, I think that Parquet is the wrong place to be defending against (potential) gaps in functionality of non-project libraries like chrono.

Comment (houqp) [reply rust/parquet/src/record/api.rs:905]: I agree that it's better to implement timestamp as signed, not only does it address the support for time before 1970, but also make it much easier to work with subtractions.

Comment (paddyhoran) [reply rust/parquet/src/record/api.rs:905]: Yea, I agree with the direction.  I was just wondering if we were relying on this assumption (non-negative milliseconds) elsewhere in the codebase.  I'll merge this soon if there are no other comments.

**Review by nevi-me — APPROVED** [review]

**Comment by nevi-me** [comment]
I've merged this, I've been having computer issues so haven't been able to do much work

**Pull request closed without merging**